src/v/datalake/translation/scheduling.h

```diff
@@ -105,6 +104,20 @@ class reservations_tracker {
 
     virtual size_t allocated_memory() const = 0;
 
+    /**
+     * Returns a reservation of disk space. The reservation may be any size,
+     * with the passed-in size being a hint. If more units are reserved than are
+     * needed, then release the unused units if they will not be used in the
+     * near term (e.g. a translator leaving the running state).
+     *
+     * Returning the unused units is important because disk space reservations
+     * are held across all translator states, including the idle state. Because
+     * of this persistence, even small amounts of unused reservation can add up
+     * to large reservations across hundreds or thousands of translators.
+     */
+    virtual ss::future<reservation> reserve_disk(size_t, ss::abort_source&) = 0;
```

Comment:
nit: do you expect the input hint to actually be used in the near future? If not, can we remove it and call this some `reserve_disk_block(ss::abort_source&)`? It seems like an easy mistake to make to expect the hint to actually be used and unintentionally rely on behavior that isn't true

Reply:
Ill put in the next PR